src/appleseed/foundation/math/bvh/bvh_middlepartitioner.h

```diff
@@ -111,7 +111,7 @@ inline size_t MiddlePartitioner<AABBVector>::partition(
 
     // In case there are multiple bboxes perfectly overlapping, return the next one.
     if (pivot == 0)
-        pivot = (end - begin) / 2;
+        pivot = (begin + end) / 2;
```

Comment:
My advice here: cover this with a unit test (that would fail with the previous code) :)